File: app/src/main/java/utils/router/RouterUtils.java
package utils.router;

import com.therouter.TheRouter;
import com.tristana.sandroid.R;

/**
 * @author koala
 * @version 1.0
 * @date 2022/4/30 14:50
 * @description
 */
public class RouterUtils {
    public static void route(String path) {
        TheRouter
                .build(path)
                .navigation();
    }

    public static void routeWithDirect(String path, String direct) {
        TheRouter
                .build(path)
                .withString("direct", direct)
                .navigation();
    }
}

File: app/src/main/java/utils/router/RouterUtils.kt
package utils.router

import com.therouter.TheRouter.build

/**
 * @author koala
 * @version 1.0
 * @date 2022/4/30 14:50
 * @description
 */
object RouterUtils {
    fun route(path: String?) {
        build(path)
            .navigation()
    }

    fun routeWithDirect(path: String?, direct: String?, extra: String? = null) {
        build(path)
            .withString("direct", direct)
            .withString("extra", extra)
            .navigation()
    }
}